pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -457,6 +446,17 @@ func (r *ReconcileNetworkAddonsConfig) trackDeployedObjects(objs []*unstructured
 	r.statusManager.SetFromPods()
 }
 
+func extendContainerStats(containersStats *[]opv1alpha1.Container, parentKind string, parentName string, containers []v1.Container) {
```

Comment:
I'm not very fond of the idea of creating an empty array in the caller side, and then filling it up in the function; what do you think of creating it within, and returning the array ? 

I just think that the caller shouldn't be bothered to be creating those structs beforehand. Feel free to disagree, it's mostly cosmetics; I can `lgtm` if you don't want to do it.

Reply:
What do you suggest to do instead? Return one list for containers, another for init contaiers and join them at the caller side?

Reply:
that or 

```golang
func extendContainerStats(containersStats *[]opv1alpha1.Container, parentKind string, parentName string, containers []v1.Container...)
```

and return a single list w/ everything. Is the above possible ?

Reply:
Check it now